File: Online-Market-Place-Server/Src/Core/OnlineMarketPlace.Domain/DatabaseEntities/ObjectForSale.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace OnlineMarketPlace.Domain
{
    public class ObjectForSale : Entity
    {
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }

        public User User { get; set; }
        public int UserId { get; set; }

        public ObjectType ObjectType { get; set; }
        public int ObjectTypeId { get; set; }

        public ICollection<AttributeValue> AttributeValues { get; set; }
    }
}

File: Online-Market-Place-Client/src/app/models/object-for-sale.ts
import { AttributeType } from "./attribute-type";
import { AttributeValue } from "./attribute-value";
import { ObjectType } from "./object-type";
import { User } from "./user";

export class ObjectForSale {
    id: number;
    price: number;
    user: User;
    objectType: ObjectType;
    attributeValues: AttributeValue[];
}